src/main.ts

```diff
@@ -103,6 +103,40 @@ export const append = <E, A, B>(
   );
 };
 
+/**
+ * Fold is the only way to attempt to safely extract the data,
+ * because it force you to consider the four variation of the dataway state.
+ * The following example demostrate how usefull this is when trying to render a
+ * search result, we may want to notify the user why no result show up on his screen
+ * 1. The search could not have been initiated by the user (he didn't click on the search button),
+ *      a loading visual hint coulb be displayed
+ * 2. The search could be initiated but not done yet (search can take time),
```

Comment:
`notAsked` et le `loading` sont inversés je pense

```suggestion
 * 1. The search could be initiated but not done yet (he didn't click on the search button),
 *      nothing should be happening
 * 2. The search has been initiated by the user (search can take time),
 *      a loading visual hint could be displayed
 * 3. The search could end up in error (timeout, connectivity issue or plain server crash)
 * 4. The search can success and we want to display the result to the user.
```

Reply:
c'est fixé